### DISCUSSION TITLE
Is there a way to highlight those overridden methods?

### DISCUSSION THREAD

**Discussion by xxc-zsz** [microsoft/pylance-release#3465] [Enhancement]
When I override one or several methods of the parent class in a subclass, is it possible to highlight those overridden methods?

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by judej** [comment]
FYI, There is a PEP regarding method override: https://peps.python.org/pep-0698/

**Comment by debonte** [comment]
@xxc-zsz, does single-clicking on the overridden method's name give you the behavior that you want?

For example, given the code below, if you single-click on the `execute` method name in `Derived`, the `execute` method in `Base` is also highlighted. But other similar methods like `Base.execute2` and `OtherClass.execute` are not highlighted.

Note that double-click apparently does a string search, so all instances of the string "execute" within the file would be highlighted. So be sure to single-click.

I noticed that if the parameter list of the two methods (`Derived.execute` and `Base.execute`) do not match, the highlight still occurs. Which I guess is a bug.

```python
import abc


class Base(abc.ABC):
    @abc.abstractmethod
    def execute(self):
        pass

    def execute2(self):
        pass

class OtherClass(abc.ABC):
    @abc.abstractmethod
    def execute(self):
        pass


class Derived(Base):
    def execute(self):
        pass
```

**Comment by xxc-zsz** [comment]
Thank you very much for your reply. 
Yes, the code you provided can do the highlighting on mouse click. But this will show all the methods inherited from a class.

![image](https://user-images.githubusercontent.com/108051416/195239565-4335f870-9cf8-4b2e-95bd-ad538e8efe43.png)
![image](https://user-images.githubusercontent.com/108051416/195237201-d3fc228d-d301-4593-a756-eac14040e7fc.png)

I can't rate whether this is good or bad. But the main purpose of my question is if there are multiple methods in the parent class and the child class only overrides some of them. Is there a way to highlight those overridden methods? Is it overcomplicated if I need my one-click method?

Any ideas are appreciated.

**Reply by debonte** [reply]
> if there are multiple methods in the parent class and the child class only overrides some of them. Is there a way to highlight those overridden methods?

@xxc-zsz, I'm confused about what your goal is. Are you wanting to highlight the methods that are **not** overridden?

Maybe you could create a mocked up "screenshot" of the highlighting behavior that you want? Or alternatively provide a code sample of "multiple methods in the parent class and the child class only overrides some of them" and be clear about what things you want highlighted?
